harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/localfilesystem/LocalFilesystemHarvester.java

```diff
@@ -123,25 +124,23 @@ private HarvestResult align(Path root) throws Exception {
 			}
 		}
 		result = visitor.getResult();
-		List<String> idsForHarvestingResult = visitor.getIdsForHarvestingResult();
+		List<Integer> idsForHarvestingResult = visitor.getIdsForHarvestingResult();
 		if (!params.nodelete) {
 			//
 			// delete locally existing metadata from the same source if they
 			// were
 			// not in this harvesting result
 			//
-            List<Metadata> existingMetadata = context.getBean(MetadataRepository.class).findAllByHarvestInfo_Uuid(params.getUuid());
-            for (Metadata existingId : existingMetadata) {
+		    List<Integer> existingMetadata = context.getBean(MetadataRepository.class).findAllIdsBy(MetadataSpecs.hasHarvesterUuid(params.getUuid()));
+            for (Integer existingId : existingMetadata) {
 
 				if (cancelMonitor.get()) {
 					return this.result;
 				}
-
-				String ex$ = String.valueOf(existingId.getId());
-				if (!idsForHarvestingResult.contains(ex$)) {
-					log.debug("  Removing: " + ex$);
-					dataMan.deleteMetadata(context, ex$);
-					result.locallyRemoved++;
+                if (!idsForHarvestingResult.contains(existingId)) {
+                    log.debug("  Removing: " + existingId);
+                    dataMan.deleteMetadata(context, existingId.toString());
```

Comment:
Not sure: imagine we need to bulk delete +250k records, the underlying query will be a:
```SQL
 DELETE FROM blah WHERE id = ... OR id = ... OR id = ...
```

I already encountered issues with PostGreSQL, ending up by the following error log

```
2015-02-10 17:55:09 CET ERROR:  stack depth limit exceeded
2015-02-10 17:55:09 CET HINT:  Increase the configuration parameter "max_stack_depth" (currently 2048kB), after ensuring the platform's stack depth limit is adequate.
```

And closing the client connection